addons/website_sale_delivery/controllers/main.py

```diff
@@ -195,7 +197,42 @@ def express_checkout_process_shipping_address(self, partial_shipping_address):
                 order_sudo.currency_id,
             ),
         } for carrier in order_sudo._get_delivery_methods()],
-        key=lambda carrier: carrier['minorAmount'])
+            key=lambda carrier: carrier['minorAmount'])
+
+    @http.route('/shop/access_point/set', type='json', auth='public', methods=['POST'], csrf=False, website=True, sitemap=False)
+    def set_access_point(self, access_point_encoded):
+        order = request.website.sale_get_order()
+        if hasattr(order.carrier_id, order.carrier_id.delivery_type + '_use_locations'):
+            use_location = getattr(order.carrier_id, order.carrier_id.delivery_type + '_use_locations')
+            access_point = use_location and access_point_encoded or None
+            order.write({'access_point_address': access_point})
+
+    @http.route('/shop/access_point/get', type='json', auth='public', csrf=False, website=True, sitemap=False)
+    def get_access_point(self):
+        order = request.website.sale_get_order()
+        if order.carrier_id.delivery_type and hasattr(order, '_get_access_point_address'):
+            order_location = getattr(order, '_get_access_point_address')()
+            if order_location:
+                address = order_location['address']
+                return {order.carrier_id.delivery_type + '_access_point': address}
+        return {}
+
+    @http.route('/shop/access_point/close_locations', type='json', auth='public', csrf=False, website=True, sitemap=False)
+    def get_close_locations(self):
+        order = request.website.sale_get_order()
+        try:
+            if hasattr(order.carrier_id, '_' + order.carrier_id.delivery_type + '_get_close_locations'):
+                close_locations = getattr(order.carrier_id, '_' + order.carrier_id.delivery_type + '_get_close_locations')(order.partner_shipping_id)
+                partner_address = order.partner_shipping_id
+                inline_partner_address = ' '.join((part or '') for part in [partner_address.street, partner_address.street2, partner_address.zip, partner_address.country_id.code])
+                if len(close_locations) > 0:
+                    for location in close_locations:
+                        location['stringified'] = json.dumps(location)
```

Comment:
stringified is a bit strange as name. You could try something more explicit on the content